### PR TITLE
Fix of rotation and pinch gesture on android

### PR DESCRIPTION
**Pull request by coado**
## Description

Fixing rotation and pinch gesture to continue to work after releasing one of a key pointers.

## Test plan

I've checked if each example on the test app still works.

### CODE REVIEW

File: android/src/main/java/com/swmansion/gesturehandler/core/RotationGestureHandler.kt [hunk 1]
```diff
@@ -59,7 +59,8 @@ class RotationGestureHandler : GestureHandler<RotationGestureHandler>() {
       anchorY = point.y
     }
     if (sourceEvent.actionMasked == MotionEvent.ACTION_UP) {
-      if (state == STATE_ACTIVE) {
+      var activePointers = sourceEvent.pointerCount
+      if (state == STATE_ACTIVE && activePointers == 0) {
```
Comment (m-bert) [inline android/src/main/java/com/swmansion/gesturehandler/core/RotationGestureHandler.kt:63]: Wouldn't this be enough?

```suggestion
      if (state == STATE_ACTIVE && sourceEvent.pointerCount == 0) {
```

File: android/src/main/java/com/swmansion/gesturehandler/core/RotationGestureDetector.kt [hunk 2]
```diff
@@ -111,11 +111,22 @@ class RotationGestureDetector(private val gestureListener: OnRotationGestureList
...
+              pointerIds[0] = pointerIds[1]
+            }
+            pointerIds[1] = MotionEvent.INVALID_POINTER_ID
+            isInProgress = false
+          }
+        } else {
+          // all key pointers are up
+          if (pointerId == pointerIds[0]) {
+            gestureListener?.onRotationEnd(this)
+          }
```
Comment (j-piasecki) [inline android/src/main/java/com/swmansion/gesturehandler/core/RotationGestureDetector.kt:129]: This if-sandwich can be simplified 😃 
```suggestion
        // All key pointers are up
        if (!isInProgress && pointerId == pointerIds[0]) {
          gestureListener?.onRotationEnd(this)
        }

        // One of the key pointers is up
        if (isInProgress && pointerIds.contains(pointerId)) {
          if (pointerId == pointerIds[0]) {
            pointerIds[0] = pointerIds[1]
          }
          pointerIds[1] = MotionEvent.INVALID_POINTER_ID
          isInProgress = false
        }
```

File: android/src/main/java/com/swmansion/gesturehandler/core/PinchGestureHandler.kt [hunk 3]
```diff
@@ -74,7 +74,7 @@ class PinchGestureHandler : GestureHandler<PinchGestureHandler>() {
     if (sourceEvent.actionMasked == MotionEvent.ACTION_POINTER_UP) {
       activePointers -= 1
     }
-    if (state == STATE_ACTIVE && activePointers < 2) {
+    if (state == STATE_ACTIVE && activePointers == 0) {
```
[the remark this answers is not in the document]
Comment (coado) [reply android/src/main/java/com/swmansion/gesturehandler/core/PinchGestureHandler.kt:77]: That's correct, this if statement is useless at this point.

[the remark this answers is not in the document]
Comment (j-piasecki) [reply android/src/main/java/com/swmansion/gesturehandler/core/PinchGestureHandler.kt:77]: I think it can be simply changed to something like this:

```
if action is ACTION_UP:
  if (state == ACTIVE):
    end()
  else
    fail()
```

File: android/src/main/java/com/swmansion/gesturehandler/core/RotationGestureDetector.kt [hunk 4]
```diff
@@ -111,11 +111,22 @@ class RotationGestureDetector(private val gestureListener: OnRotationGestureList
...
+        }
+
+        // One of the key pointers is up
+        if (isInProgress && pointerIds.contains(pointerId)) {
+          if (pointerId == pointerIds[0]) {
+            pointerIds[0] = pointerIds[1]
+          }
+          pointerIds[1] = MotionEvent.INVALID_POINTER_ID
+          isInProgress = false
+        }
```
Comment (j-piasecki) [inline android/src/main/java/com/swmansion/gesturehandler/core/RotationGestureDetector.kt:129]:
```suggestion
```

I think I left one curly boi too much in the previous suggestion

File: android/src/main/java/com/swmansion/gesturehandler/core/RotationGestureHandler.kt [hunk 5]
```diff
@@ -59,7 +59,7 @@ class RotationGestureHandler : GestureHandler<RotationGestureHandler>() {
       anchorY = point.y
     }
     if (sourceEvent.actionMasked == MotionEvent.ACTION_UP) {
-      if (state == STATE_ACTIVE) {
+      if (state == STATE_ACTIVE && sourceEvent.pointerCount == 0) {
```
Comment (j-piasecki) [inline android/src/main/java/com/swmansion/gesturehandler/core/RotationGestureHandler.kt:62]: Is this change necessary? I think it's the same as with pinch - `pointerCount` may never be equal to 0.

File: android/src/main/java/com/swmansion/gesturehandler/core/PinchGestureHandler.kt [hunk 6]
```diff
@@ -74,10 +74,13 @@ class PinchGestureHandler : GestureHandler<PinchGestureHandler>() {
     if (sourceEvent.actionMasked == MotionEvent.ACTION_POINTER_UP) {
       activePointers -= 1
```
Comment (j-piasecki) [inline android/src/main/java/com/swmansion/gesturehandler/core/PinchGestureHandler.kt:75]: I don't think `activePointers` is being used anymore, can you remove it?

**Comment by j-piasecki** [comment]
Can you also update the title of the PR?

**Review by j-piasecki — APPROVED** [review]
